src/lib.rs

```diff
@@ -262,20 +262,125 @@ impl<T> StableVec<T> {
     /// This method uses exactly the same deletion strategy as
     /// [`remove()`](#method.remove).
     ///
+    /// # Example
+    ///
+    /// ```
+    /// # use stable_vec::StableVec;
+    /// let mut sv = StableVec::from(&[1, 2, 3]);
+    /// assert_eq!(sv.pop(), Some(3));
+    /// assert_eq!(sv.into_vec(), vec![1, 2]);
+    /// ```
+    ///
     /// # Note
     ///
     /// This method needs to find index of the last valid element. Finding it
     /// has a worst case time complexity of O(n). If you already know the
     /// index, use [`remove()`](#method.remove) instead.
     pub fn pop(&mut self) -> Option<T> {
-        let last_index = self.deleted
-            .iter()
-            .enumerate()
-            .rev()
-            .find(|&(_, deleted)| !deleted)
-            .map(|(i, _)| i)
-            .unwrap_or(0);
-        self.remove(last_index)
+        self.find_last_index().and_then(|index| self.remove(index))
+    }
+
+    /// Removes and returns the first element from this collection, or `None` if
+    /// it's empty.
+    ///
+    /// This method uses exactly the same deletion strategy as
+    /// [`remove()`](#method.remove).
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use stable_vec::StableVec;
+    /// let mut sv = StableVec::from(&[1, 2, 3]);
+    /// assert_eq!(sv.pop_first(), Some(1));
+    /// assert_eq!(sv.into_vec(), vec![2, 3]);
+    /// ```
+    ///
+    /// # Note
+    ///
+    /// This method needs to find index of the first valid element. Finding it
+    /// has a worst case time complexity of O(n). If you already know the
+    /// index, use [`remove()`](#method.remove) instead.
+    pub fn pop_first(&mut self) -> Option<T> {
```

Comment:
Maybe rather `remove_first`? Maybe it's also worth renaming `pop` to `remove_last`?

Reply:
Sounds sensible to me.

Reply:
It's probably better to implement `pop` as a call to `remove_last` and keep it to avoid unnecessary breaking changes. You can make it deprecated and remove later in some major release, if you want.

Reply:
Sounds good!